packaging/bundle-validation/ci_run.sh

```diff
@@ -110,95 +112,116 @@ fi
+  mkdir -p $TMP_DATA_DIR/stocks/data
+  cp ${GITHUB_WORKSPACE}/docker/demo/data/*.json $TMP_DATA_DIR/stocks/data/
+  cp ${GITHUB_WORKSPACE}/docker/demo/config/schema.avsc $TMP_DATA_DIR/stocks/
+
+  # build docker image
+  cd ${GITHUB_WORKSPACE}/packaging/bundle-validation || exit 1
+  docker build \
+  --build-arg HADOOP_VERSION=$HADOOP_VERSION \
+  --build-arg HIVE_VERSION=$HIVE_VERSION \
+  --build-arg DERBY_VERSION=$DERBY_VERSION \
+  --build-arg FLINK_VERSION=$FLINK_VERSION \
+  --build-arg SPARK_VERSION=$SPARK_VERSION \
+  --build-arg SPARK_HADOOP_VERSION=$SPARK_HADOOP_VERSION \
+  --build-arg CONFLUENT_VERSION=$CONFLUENT_VERSION \
+  --build-arg KAFKA_CONNECT_HDFS_VERSION=$KAFKA_CONNECT_HDFS_VERSION \
+  --build-arg IMAGE_TAG=$IMAGE_TAG \
+  -t hudi-ci-bundle-validation:$IMAGE_TAG \
+  .
 
-ls -l $TMP_JARS_DIR
-
-# Copy test dataset
-TMP_DATA_DIR=/tmp/data/$(date +%s)
-mkdir -p $TMP_DATA_DIR/stocks/data
-cp ${GITHUB_WORKSPACE}/docker/demo/data/*.json $TMP_DATA_DIR/stocks/data/
-cp ${GITHUB_WORKSPACE}/docker/demo/config/schema.avsc $TMP_DATA_DIR/stocks/
-
-# build docker image
-cd ${GITHUB_WORKSPACE}/packaging/bundle-validation || exit 1
-docker build \
---build-arg HADOOP_VERSION=$HADOOP_VERSION \
---build-arg HIVE_VERSION=$HIVE_VERSION \
---build-arg DERBY_VERSION=$DERBY_VERSION \
---build-arg FLINK_VERSION=$FLINK_VERSION \
---build-arg SPARK_VERSION=$SPARK_VERSION \
---build-arg SPARK_HADOOP_VERSION=$SPARK_HADOOP_VERSION \
---build-arg CONFLUENT_VERSION=$CONFLUENT_VERSION \
---build-arg KAFKA_CONNECT_HDFS_VERSION=$KAFKA_CONNECT_HDFS_VERSION \
---build-arg IMAGE_TAG=$IMAGE_TAG \
--t hudi-ci-bundle-validation:$IMAGE_TAG \
-.
-
-# run validation script in docker
-docker run -v $TMP_JARS_DIR:/opt/bundle-validation/jars -v $TMP_DATA_DIR:/opt/bundle-validation/data \
-  -i hudi-ci-bundle-validation:$IMAGE_TAG bash validate.sh $JAVA_RUNTIME_VERSION
+  # run validation script in docker
+  docker run -v $TMP_JARS_DIR:/opt/bundle-validation/jars -v $TMP_DATA_DIR:/opt/bundle-validation/data \
+    -i hudi-ci-bundle-validation:$IMAGE_TAG bash validate.sh $JAVA_RUNTIME_VERSION
+else
+  echo "Running Docker test for Java 17, skipping jar copying"
+  # build docker image
+  cd ${GITHUB_WORKSPACE}/packaging/bundle-validation || exit 1
+  docker build \
+  --build-arg HADOOP_VERSION=$HADOOP_VERSION \
+  --build-arg HIVE_VERSION=$HIVE_VERSION \
+  --build-arg DERBY_VERSION=$DERBY_VERSION \
+  --build-arg FLINK_VERSION=$FLINK_VERSION \
+  --build-arg SPARK_VERSION=$SPARK_VERSION \
+  --build-arg SPARK_HADOOP_VERSION=$SPARK_HADOOP_VERSION \
+  --build-arg CONFLUENT_VERSION=$CONFLUENT_VERSION \
+  --build-arg KAFKA_CONNECT_HDFS_VERSION=$KAFKA_CONNECT_HDFS_VERSION \
+  --build-arg IMAGE_TAG=$IMAGE_TAG \
+  -t hudi-ci-bundle-validation:$IMAGE_TAG \
+  .
+
+  docker run -v ${GITHUB_WORKSPACE}:/opt/bundle-validation/docker-test \
+    -i hudi-ci-bundle-validation:$IMAGE_TAG bash docker_test_java17.sh $JAVA_RUNTIME_VERSION $SPARK_PROFILE $SCALA_PROFILE
```

Comment:
Either we consolidate the Java 17 tests into bundle validation (conditionally) or we create a new script for Java 17 test only.

Reply:
Makes sense.